File: 2003/IntroToVBNETPart2/Presentation/ConvertEnum.cs
using System ;

public class ConvertEnum
{
 public static void Main ( )
 {
  string [] days = new string [] { "Sunday", "Monday", "Tuesday", 
   "Wednesday", "Thursday", "Friday", "Saturday" } ;
  int i ;
  DayOfWeek day = DayOfWeek.Monday ; // Initialize, otherwise compiler complains about uninitialized variable
  for ( i = 0 ; i <= days.Length - 1 ; i++ )
  {
   day = (DayOfWeek)  System.Enum.Parse ( day.GetType(), days [ i ] ) ;
   Console.WriteLine ( "{0} {1}", day, (int) day ) ;
  }
 }
}

File: 2003/IntroToVBNETPart2/Presentation/ConvertEnum.vb
Imports System

Public Class ConvertEnum
	Public Shared Sub Main ( )
		Dim days () As String = New String () { "Sunday", "Monday", "Tuesday", _
			"Wednesday", "Thursday", "Friday", "Saturday" }
		Dim i As Integer
		Dim day As DayOfWeek
		For i = 0 To days.Length - 1
			day = CType( System.Enum.Parse(day.GetType(), days(i) ), DayOfWeek)
			Console.WriteLine ( "{0} {1}", day, CInt(day) )
		Next i
	End Sub
End Class